File: src/tests/fidl/source_compatibility/enum-flexible-strict/llcpp/step_00_before.cc
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/fidl.test.enumflexiblestrict/cpp/wire.h>  // nogncheck
namespace fidl_test = fidl_test_enumflexiblestrict;

// [START contents]
fidl_test::wire::Color complement(fidl_test::wire::Color color) {
  if (color.IsUnknown()) {
    return color;
  }
  switch (color) {
    case fidl_test::wire::Color::kRed:
      return fidl_test::wire::Color::kBlue;
    case fidl_test::wire::Color::kBlue:
      return fidl_test::wire::Color::kRed;
    default:
      return color;
  }
}
// [END contents]

int main(int argc, const char** argv) { return 0; }

File: src/tests/fidl/source_compatibility/enum-flexible-strict/dart/step_00_before.dart
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// TODO(https://fxbug.dev/84961): Fix null safety and remove this language version.
// @dart=2.9

import 'package:fidl_fidl_test_enumflexiblestrict/fidl_async.dart' as fidllib;

// [START contents]
fidllib.Color complement(fidllib.Color color) {
  if (color.isUnknown()) {
    return color;
  }
  switch (color) {
    case fidllib.Color.blue:
      return fidllib.Color.red;
    case fidllib.Color.red:
      return fidllib.Color.blue;
    default:
      return null;
  }
}

// [END contents]
